### ISSUE TITLE
The dagster-pandera integration cannot handle `Annotated` types

### ISSUE THREAD

**Issue by Midnighter** [dagster-io/dagster#23000]
### Dagster version

1.7.13

### What's the issue?

The dagster-pandera integration does not work `Annotated` types, but raises an error.

```
ValueError: no signature found for builtin type <class 'str'>
```

### What did you expect to happen?

The below example is valid pandera code, so I expected it to be possible to convert to a dagster type.

### How to reproduce?

```py
from typing import Annotated

import pandera as pa
from dagster_pandera import pandera_schema_to_dagster_type
from pandera.typing import Series


class MyFrame(pa.DataFrameModel):

    something: Series[
        Annotated[
            str,
            pa.Field(description="Collected info on something."),
        ]
    ]


MyFrameDagsterType = pandera_schema_to_dagster_type(MyFrame)
```

Instead the following succeeds:

```py
import pandera as pa
from dagster_pandera import pandera_schema_to_dagster_type
from pandera.typing import Series


class MyFrame(pa.DataFrameModel):

    something: Series[str] = pa.Field(description="Collected info on something.")


MyFrameDagsterType = pandera_schema_to_dagster_type(MyFrame)
```

### Deployment type

Local

### Deployment details

The above mini example can be dropped in any file and run with Python directly.

### Message from the maintainers

Impacted by this issue? Give it a 👍! We factor engagement into prioritization.

**Comment by NGHades** [comment]
I'm working on this!

**Comment by kevJ711** [comment]
I’m currently working on this issue with my team (@NGHades and @KViruz2750 ) and wanted to confirm our current understanding of the issue:

**TL;DR**: Using `Annotated` types with `pandera_schema_to_dagster_type()` raises a `ValueError` because metadata isn’t being handled correctly in Pandera’s `build_schema()`. The failure traces through `get_dtype_kwargs()` to Python’s `inspect.signature()`.

### Summary of the Issue  

When using `pandera_schema_to_dagster_type()` with a DataFrame model (`MyFrame`) that relies on `Annotated` types from Python’s `typing` module, we encounter a `ValueError`. This occurs during the [`to_schema()`](https://github.com/dagster-io/dagster/blob/master/python_modules/libraries/dagster-pandera/dagster_pandera/__init__.py#L137) call, which is supposed to convert the DataFrame model into a Pandera `DataFrameSchema`.  

- The error seems to originate in [`build_schema()`](https://github.com/unionai-oss/pandera/blob/main/pandera/api/dataframe/model.py#L253) (inside the Pandera OSS) when handling metadata in `pa.Field`.  
- Replacing `Annotated` with `Any` or `Optional` works fine, but using `Annotated` with metadata consistently raises errors.  
- The failure appears after the `multiindex_kwargs` section of `build_schema()`. Tracing further, the issue seems tied to `get_dtype_kwargs()` and eventually [`inspect.signature()`](https://github.com/unionai-oss/pandera/blob/main/pandera/api/dataframe/model.py#L61)  

**Hypothesis:** Metadata in Annotated types is not being correctly read or written during the DataFrameModel → DataFrameSchema conversion. Could the root cause be how inspect.signature() handles Annotated, or is there another integration issue between Pandera and Dagster at play?

---

### Detailed Debugging Process  

#### Where the Error Happens  
- **Function:** [`pandera_schema_to_dagster_type()`](https://github.com/dagster-io/dagster/blob/master/python_modules/libraries/dagster-pandera/dagster_pandera/__init__.py](https://github.com/dagster-io/dagster/blob/master/python_modules/libraries/dagster-pandera/dagster_pandera/__init__.py#L94))  
- Inside this function, calling `to_schema()` causes the program to terminate.  
- `to_schema()` is supposed to convert a class-based DataFrame model into a Pandera `DataFrameSchema`.  
- **Relevant code:** [`pandera/api/dataframe/model.py`](https://github.com/unionai-oss/pandera/blob/main/pandera/api/dataframe/model.py)  

---

#### Initial Investigation  
- Inside `to_schema()`, the call to `build_schema()` seems to be the problem.  
- `build_schema()` fails specifically when handling metadata in `pa.Field`.  

Example error:  
```python
(<pandera.api.dataframe.model_components.FieldInfo("None") object at 0x77678e3d53b0>,)
```
#### Tests We Tried Replacing
- Annotated with Any or Optional types → no error. 
- Changing pa.Field(description="...") into a string literal (Annotated[str, "…"]) → error persisted. 
- Using Annotated[str] (without metadata) → new error stating Annotated requires 2 arguments. 
- Added print statements: Confirmed the issue occurs inside build_schema() after the multiindex_kwargs segment. 
- Commenting out the if name.startswith("multiindex_") block allowed print statements to execute afterward.

The error appears to originate here, though it definitely due to how Pandera handles Annotated types. I created an issue post within the Pandera OSS [here](https://github.com/unionai-oss/pandera/issues/2110)

**Comment by kevJ711** [comment]
@danielgafni

**Comment by danielgafni** [comment]
Hey guys

1. I am not maintaining `dagster-pandera`, you should find another reviewer for this PR. @garethbrickman should be able to help here. 
2. An advise regarding our report: I can see you went though some thorough debugging and reported a bunch of info -- which is great, and I recognize your efforts here, but the report is not very accessible in the current format. 
Please use Markdown formatting: reference files with `[]()` syntax, like [__init__.py](python_modules/libraries/dagster-pandera/dagster_pandera/__init__.py), wrap code blocks in backticks (`), provide concrete stack traces and logs instead of describing print statements. Some of your references are broken, such as model.py (of course it points to a website instead of pandera code). All of this make following the explanation very hard.

 I think providing a more clear report might get the issue or PR reviewed faster!
